google-cloud-dialogflow/src/test/java/com/google/cloud/dialogflow/v2/it/ITSystemTest.java

```diff
@@ -0,0 +1,324 @@
+            .setDisplayName(ENTITY_NAME)
+            .setKind(EntityType.Kind.KIND_LIST)
+            .addEntities(ENTITY)
+            .build();
+    CreateEntityTypeRequest entityTypeRequest =
+        CreateEntityTypeRequest.newBuilder()
+            .setParent(PROJECT_AGENT_NAME.toString())
+            .setEntityType(createEntityType)
+            .build();
+    entityType = entityTypesClient.createEntityType(entityTypeRequest);
+    entityName =
+        entityType.getName().substring(entityType.getName().lastIndexOf("/")).replace("/", "");
+    entityTypeName = EntityTypeName.of(PROJECT_ID, entityName);
+
+    /* create intents */
+    intentsClient = IntentsClient.create();
+    Intent createIntent =
+        Intent.newBuilder()
+            .setDisplayName(INTENT_NAME)
+            .addEvents(EVENT_NAME)
+            .setAction(ACTION_NAME)
+            .build();
+    CreateIntentRequest intentRequest =
+        CreateIntentRequest.newBuilder()
+            .setParent(PROJECT_AGENT_NAME.toString())
+            .setIntent(createIntent)
+            .build();
+    intent = intentsClient.createIntent(intentRequest);
+    intentId = intent.getName().substring(intent.getName().lastIndexOf("/")).replace("/", "");
+    intentName = IntentName.of(PROJECT_ID, intentId);
+
+    /* create session */
+    sessionsClient = SessionsClient.create();
+
+    /* create context */
+    contextsClient = ContextsClient.create();
+    Context createContext =
+        Context.newBuilder()
+            .setName(CONTEXT_NAME.toString())
+            .setLifespanCount(LIFE_SPAN_COUNT)
+            .build();
+    CreateContextRequest request =
+        CreateContextRequest.newBuilder()
+            .setParent(SESSION_NAME.toString())
+            .setContext(createContext)
+            .build();
+    context = contextsClient.createContext(request);
+  }
+
+  @AfterClass
+  public static void tearDown() {
+
+    /* delete context */
+    DeleteContextRequest deleteContextRequest =
+        DeleteContextRequest.newBuilder().setName(CONTEXT_NAME.toString()).build();
+    contextsClient.deleteContext(deleteContextRequest);
+    contextsClient.close();
+
+    /* delete session */
+    sessionsClient.close();
+
+    /* delete intents */
+    DeleteIntentRequest deleteIntentRequest =
+        DeleteIntentRequest.newBuilder().setName(intentName.toString()).build();
+    intentsClient.deleteIntent(deleteIntentRequest);
+    intentsClient.close();
+
+    /* delete entity */
+    DeleteEntityTypeRequest deleteEntityTypeRequest =
+        DeleteEntityTypeRequest.newBuilder().setName(entityTypeName.toString()).build();
+    entityTypesClient.deleteEntityType(deleteEntityTypeRequest);
+    entityTypesClient.close();
+
+    /* delete agent */
+    DeleteAgentRequest deleteAgentRequest =
+        DeleteAgentRequest.newBuilder().setParent(PROJECT_NAME.toString()).build();
+    agentsClient.deleteAgent(deleteAgentRequest);
+    agentsClient.close();
```

Comment:
We don't want to delete the agent - our smoke test and samples (in the future) will need a default agent present to function. If we create a new agent in the `@Before`, we could delete this one, but I believe all the dialogflow agents are shared in a single organization so we may not want to repeatedly create/delete agents.

Reply:
Done.